server_modules/utils.js

```diff
@@ -1,16 +1,10 @@
-exports.root = __dirname + '/../';
-
-exports.sendError = function(res) {
-    return function(err) {
-        res.status(err.status).send(err.message);
-    }
-}
-
-if (!String.prototype.format) {
-    String.prototype.format = function() {
-        var args = arguments;
-        return this.replace(/{(\d+)}/g, function(match, number) { 
-            return typeof args[number] !== 'undefined' ? args[number] : match;
-        });
-    };
-}
+var log = require('./log').log;
+
+exports.root = __dirname + '/../';
+
+exports.sendError = function(res) {
```

Comment:
This is a tricky function to use: it is very easy to call it different from what you expect (i.e. the error object), see e.g. my comment in `file_system.js`.

Additionally, for security reasons it is typically frowned upon to directly return a server-side error as-is to the client. You may reveal all kinds of internal data.

So, better to change this to e.g. `function sendError(res, err) { ... }`
Then use it as:
```
.catch((err) => utils.sendError(res, new Error("failed to save foo.txt")))
```